src/lib/error-handler/index.ts

```diff
@@ -37,22 +61,22 @@ export const errorHandler = function errorHandler(
   next: NextFunction
 ): ExResponse | void {
   if (err instanceof ValidateError) {
-    logger.debug(`Handled Validation Error for ${req.path}:`, err.fields)
-    const response: ValidateErrorJSON = {
+    logger.error(`Handled Validation Error for ${req.path}:`, err.fields)
+
+    return res.status(422).send({
+      ...err,
       message: 'Validation failed',
-      details: err?.fields,
-    }
-    return res.status(422).json(response)
+    })
   }
   if (err instanceof HttpResponse) {
-    logger.debug(`Bad request for ${req.path}`)
+    logger.error('Unexpected error thrown in handler: %s', err.message)
```

Comment:
some of the errors are expected? Like a `400`

Reply:
not sure if I understand this question. In this instance we do not want to log debug because this is an error even tho  'client-side' error. why we are logging as erorr? when it comes to millions of logs entries we want to be able to find all errors and trace our request back if needed. This will allow us to 'filter error.level = 40/50/30/10/etc'

The req_id and levels do help as we can filter log by level e.g. error = 50 so we can say if there was an error.level = 50 over 24hours - we do something e.g. send a slack notification so it can be investigated, or we put to messaging service marked as 'un-processes' whatever:)

Reply:
I mean that any HttpResponse error is an error we've defined so it's not unexpected like a `500`, it's an expected error, but it's only a logger message so very minor point